src/main/ruby/truffleruby/core/range.rb

```diff
@@ -97,7 +99,7 @@ def bsearch(&block)
   end
 
   private def bsearch_float(&block)
-    normalized_begin = self.begin.to_f
+    normalized_begin = Primitive.nil?(self.begin) ? -Float::INFINITY : self.begin.to_f
     normalized_end = Truffle::RangeOperations.endless?(self) ? Float::INFINITY : self.end.to_f
```

Comment:
Mixed `Primitive.nil?` and `TruffleRangeOperations.endless?`.
I think we should just get rid of `TruffleRangeOperations.endless?/beginless?` and just use `Primitive.nil?` everywhere.

I'm not entirely certain why `endless?` exists in the first place, but I _think_ it was because it used to be a primitive, which was then redefined in terms of `Primitive.nil?`. No point in keeping it now.